src/backend/distributed/utils/metadata_cache.c

```diff
@@ -933,6 +934,52 @@ HasUninitializedShardInterval(ShardInterval **sortedShardIntervalArray, int shar
 
 
 /*
+ * HasOverlappingShardInterval determines whether the given list of sorted
+ * shards has overlapping ranges.
+ */
+static bool
+HasOverlappingShardInterval(ShardInterval **shardIntervalArray,
```

Comment:
I don't think it's absolutely critical to have it, but I felt uncomfortable assuming that everyone using range partitioning was careful enough about their boundaries.  That it ends up working for append partitioned tables is a nice benefit.